addons/stock/models/stock_move.py

```diff
@@ -576,6 +576,15 @@ def name_get(self):
                 move.location_id.name, move.location_dest_id.name)))
         return res
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if 'picking_id' in vals and 'group_id' not in vals:
+                picking = self.env['stock.picking'].browse(vals['picking_id'])
+                if picking.group_id:
+                    vals['group_id'] = picking.group_id.id
+        return super().create(vals_list)
```

Comment:
could you add the same idea in `write` too ? I can think of this 
https://github.com/odoo/odoo/blob/23c805030abb89ca449cf430d5661d725f5801a8/addons/stock/models/stock_move.py#L1126 
for instance